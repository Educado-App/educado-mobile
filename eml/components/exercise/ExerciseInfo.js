import React from 'react';
import { View, Text, Image } from 'react-native';

const ExerciseInfo = ({ courseId, sectionId }) => {
  return (
    <View className='flex-row justify-between items-end px-[6%] py-[12%]'>
      <View className=''>
        <Text className='font-montserrat text-caption-small text-projectGray'>
          Course name: {courseId}
        </Text>
        <Text className='font-montserrat-bold text-body text-projectBlack'>{sectionId}</Text>
      </View>
      <View className=''>
        <Image source={require('../../assets/images/dots-horizontal.png')} />
      </View>
    </View>
  );
};

export default ExerciseInfo;
