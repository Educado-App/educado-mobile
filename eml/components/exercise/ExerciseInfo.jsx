import React from 'react';
import { View } from 'react-native';
import Text from '../general/Text';
import PropTypes from 'prop-types';

/* 
Description:	Displays the course title and section title of the exercise the student is in.
Denpendencies: 	The student must be in a exercise.
Props: 			courseTitle - The title of the course the student is in.
				sectionTitle - The title of the section the student is in.
*/

const ExerciseInfo = ({ courseTitle, sectionTitle }) => {
	return (
		<View className="items-start px-6 absolute bottom-10 z-10">
			<Text className="font-sans text-caption-small text-projectGray">
        Course name: {courseTitle}
			</Text>
			<Text className="font-sans-bold text-body text-projectBlack">{sectionTitle}</Text>
		</View>
	);
};

ExerciseInfo.propTypes = {
	courseTitle: PropTypes.string,
	sectionTitle: PropTypes.string,
};

export default ExerciseInfo;
