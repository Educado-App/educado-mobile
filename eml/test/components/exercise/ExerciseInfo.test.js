import React from 'react';
import renderer from 'react-test-renderer';
import ExerciseInfo from '../../../components/exercise/ExerciseInfo';
import AsyncStorage from '@react-native-async-storage/async-storage';

let navigated = false;

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(() => {
      navigated = true;
    }),
  }),
  useRoute: () => ({
    params: {}, // Add any route parameters your component relies on
  }),
}));

let ExerciseInfoScreen;

beforeEach(() => {
  navigated = false;
  AsyncStorage.clear();
  ExerciseInfoScreen = renderer.create(<ExerciseInfo />);
});

describe('ExerciseInfo', () => {
  it('renders ExerciseInfo correctly', () => {
    expect(ExerciseInfoScreen.toJSON()).toMatchSnapshot();
  });
});
