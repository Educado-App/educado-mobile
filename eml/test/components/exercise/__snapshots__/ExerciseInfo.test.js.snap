// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseInfo renders ExerciseInfo correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
      },
      {
        "paddingLeft": 24,
        "paddingRight": 24,
      },
      {
        "position": "absolute",
      },
      {
        "bottom": 40,
      },
      {
        "zIndex": 10,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Montserrat-Regular",
        },
        {
          "color": "#383838",
        },
        {
          "fontSize": 16,
        },
        [
          {
            "fontFamily": "Montserrat-Regular",
          },
          {
            "fontSize": 10,
          },
          {
            "color": "#A1ACB2",
          },
        ],
      ]
    }
  >
    Course name: 
  </Text>
  <Text
    style={
      [
        {
          "fontFamily": "Montserrat-Regular",
        },
        {
          "color": "#383838",
        },
        {
          "fontSize": 16,
        },
        [
          {
            "fontFamily": "Montserrat-Bold",
          },
          {
            "fontSize": 16,
          },
          {
            "color": "#383838",
          },
        ],
      ]
    }
  />
</View>
`;
